src/main/java/com/code4ro/legalconsultation/service/impl/pdf/PDFServiceImpl.java

```diff
@@ -36,4 +57,38 @@ public String readAsString(final byte[] file) {
             throw new LegalValidationException("document.parse.pdf.failed", HttpStatus.BAD_REQUEST);
         }
     }
+
+    @Override
+    public PdfHandle createPdf(@NonNull DocumentConsolidated owner, String state, MultipartFile file) {
+        Integer hash = file.hashCode();
+        if (pdfHandleRepository.existsByHash(hash)) {
+            if (pdfHandleRepository.existsByHashAndState(hash, state)) {
+                return pdfHandleRepository.findByHashAndState(hash, state);
+            } else {
+                throw new PdfAlreadyExistsException(
+                        pdfHandleMapper.map(pdfHandleRepository.findByHash(hash)));
+            }
+        }
+
+        final String uriString;
+        try {
+            uriString = storageApi.storeFile(file);
+        } catch (IOException e) {
```

Comment:
```suggestion
        } catch (IOException e) {
        e.printStackTrace();
            Logger.getLogger(PDFServiceImpl.class).warn("failed to save the pdf file", e);
            return null;
        } catch (Exception e) {
            e.printStackTrace();
            throw new RuntimeException(String.format("another exception: %s", e.getLocalizedMessage()));
        }
```
} catch (Exception e) {
            log.error("Could not store the file", e);
            throw new LegalValidationException("document.upload.pdf.failed", HttpStatus.BAD_REQUEST);
        }